src/Dictionaries/HierarchyDictionariesUtils.h

```diff
@@ -26,7 +26,10 @@ class DictionaryHierarchicalParentToChildIndex
         UInt32 end_index;
     };
 
-    explicit DictionaryHierarchicalParentToChildIndex(const HashMap<UInt64, PaddedPODArray<UInt64>> & parent_to_children_map_)
+    using Array = PODArray<UInt64, 8 * sizeof(UInt64), Allocator<false>, PADDING_FOR_SIMD - 1, PADDING_FOR_SIMD>;
```

Comment:
by default we use `initial_bytes=4096` in pod array. it might lead to really high memory consumption when arrays are almost empty but there are a lot of them.

Reply:
It would be nice to add that comment to the code just in case